topology/probes/vpp/vpp.go

```diff
@@ -2,6 +2,7 @@
 
 //go:generate go run git.fd.io/govpp.git/cmd/binapi-generator --input-file=/usr/share/vpp/api/interface.api.json --output-dir=./bin_api
 //go:generate go run git.fd.io/govpp.git/cmd/binapi-generator --input-file=/usr/share/vpp/api/vpe.api.json --output-dir=./bin_api
+//go:generate go run git.fd.io/govpp.git/cmd/binapi-generator --input-file=/usr/share/vpp/api/memif.api.json --output-dir=./bin_api
 
```

Comment:
The major issue with the design of VPP binary APIs, is that practically any change breaks backward compatibility. The binary API files generated in this PR are I suppose for 19.04 VPP. I tried to run skydive against container with VPP 19.08 and got some expected errors, such as:
```
skydive-agent_1     | 2019-10-17T10:09:29.960Z	WARN	vpp/vpp.go:649 (*Probe).Fire	vpp-93026bb9-358b-4bff-414b-6c852edd130b: time="2019-10-17T10:09:29Z" level=warning msg="Unknown message received, ID: 829"
skydive-agent_1     | 2019-10-17T10:09:27.753Z	ERROR	probes/wrapper.go:52 (*serviceManager).Start.func1	vpp-a31cbfb5-a85f-4503-5ad3-9b5225f85436: no reply received within the timeout period 1s
```
This is because any change in binary API message changes its CRC which will cause it to be refused and dropped on either VPP or GoVPP side. Similary, a newly added or removed message can shift sequence IDs of existing messages, once again breaking compatibility between client and VPP.
Although you have added the `.vppbinapi` makefile target that allows to build skydive for a custom VPP version, a single skydive binary will always be limited to only one version of VPP and might not interpret network containing multiple different containerized VPPs properly. Unfortunately, the `interface.api` that you use to dump interfaces changes pretty much with every release. For example, `SwInterfaceDump` structure has been edited (interface name is `string`, not array of bytes) in 19.08 and `probes/vpp/vpp.go` needs to be edited to reflect the change for it to compile. Actually, between `19.04` and `19.08` even the API language changed a bit (`string` type was introduced) and govpp had to be adjusted. Therefore to generate APIs for 19.08 you would also need to update the govpp dependency first. To make this even worse, in 19.08 the path to installed `*.api.json` files has changed - core APIs and plugin APIs are further separated to `core` and `plugins` sub-directories.

What we do in [ligato/vpp-agent](https://github.com/ligato/vpp-agent) to mitigate this issue, is that we hide interaction with VPP behind an interface. Then we provide implementation of that interface for the latest VPP version (top of the master branch, frequently updated) and for the last 2 releases (for example: https://github.com/ligato/vpp-agent/tree/master/plugins/vpp/ifplugin/vppcalls). Each of these implementations has its own generated binary APIs corresponding to its version. For example [in this file](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/ifplugin/vppcalls/if_vppcalls.go) we define APIs for managing VPP interfaces. Then every implementation of this API registers itself into [this map](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/ifplugin/vppcalls/if_vppcalls.go#L238). When connection to a VPP instance is made, [CheckCompatibility method is used](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/ifplugin/vppcalls/if_vppcalls.go#L248) to find and select that implementation which is compatible with the given VPP. In your case you could define API with methods like `FetchMemifs()`, `FetchInterfaces()`, `SubscribeInterfaceEvents` and define your own structures for interface metadata with fields that you actually use.

If support of multiple VPP versions is not a priority right now, then maybe at least document which version is supported out-of-the-box right now and perhaps allow to switch the `.vppbinapi` makefile target off, so that skydive can be compiled with binary APIs already generated and included in the repo, without having to have VPP installed in the host.